pkg/controller/kataconfig/openshift_reconciler.go

```diff
@@ -694,3 +715,95 @@ func (r *ReconcileKataConfigOpenShift) monitorKataConfigInstallation() (reconcil
 
 	return reconcile.Result{}, nil
 }
+
+func getConfigClientSet() (*configv1.ConfigV1Client, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", "")
+	if err != nil {
+		return nil, err
+	}
+
+	configV1Client, err := configv1.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return configV1Client, err
+}
+
+func getClusterVersion() (string, error) {
+	configV0Client, err := getConfigClientSet()
+	if err != nil {
+		return "", err
+	}
+
+	clusterversion, err := configV0Client.ClusterVersions().Get("version", metav1.GetOptions{})
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		return "", nil
+	}
+
+	return clusterversion.Status.Desired.Version, nil
+}
+
+func getSourceImageTag(sourceImage string) (string, error) {
+	imageslice := strings.Split(sourceImage, ":")
+	if len(imageslice) <= 1 {
+		return "", fmt.Errorf("bad format of sourceImage")
+	}
+	mytag := imageslice[len(imageslice)-1]
+
+	return mytag, nil
+}
+
+func getClientSet() (*kubernetes.Clientset, error) {
```

Comment:
Although it's used only in OpenShift reconciler, it's not really OpenShift specific code per say. I would suggest that leave at the common location. i.e. `kubeconfig_controller.go`. It could get useful in future for other controllers if they want to fetch the clientset to interact with API objects.

Reply:
Right, I'll move it to a common file